test/wallet-test.js

```diff
@@ -2279,4 +2289,115 @@ describe('Wallet', function() {
       assert.equal(confirmedCount, 1);
     });
   });
+
+  describe('Wallet Name Claims', function() {
+    let wallet;
+
+    const network = Network.get('regtest');
+    const workers = new WorkerPool({enabled: false});
+    const wdb = new WalletDB({network, workers});
+
+    before(async () => {
+      await wdb.open();
+      wallet = await wdb.create();
+
+      for (let i = 0; i < 3; i++) {
+        const cur = new ChainEntry(curBlock(wdb));
+        const next = new Block(nextBlock(wdb));
+        const entry = ChainEntry.fromBlock(next, cur);
+        await wdb.addBlock(entry, []);
+      }
+
+      const bal = await wallet.getBalance();
+      assert.equal(bal.tx, 0);
+      assert.equal(bal.coin, 0);
+      assert.equal(bal.unconfirmed, 0);
+      assert.equal(bal.confirmed, 0);
+      assert.equal(bal.ulocked, 0);
+      assert.equal(bal.clocked, 0);
+    });
+
+    after(async () => {
+      await wdb.close();
+    });
+
+    it('should not have any cloudflare state', async () => {
+      const rawName = Buffer.from('cloudflare', 'ascii');
+      const nameHash = rules.hashName(rawName);
+      const nameinfo = await wallet.getNameState(nameHash);
+      assert.deepEqual(nameinfo, null);
+    });
+
+    it('should confirm cloudflare CLAIM', async () => {
+      const claim = await wallet.sendFakeClaim('cloudflare');
+      assert(claim);
+
+      const tx = claim.toTX(network, wdb.state.height + 1);
+
+      const cur = new ChainEntry(curBlock(wdb));
+      const next = new Block(nextBlock(wdb));
+      const entry = ChainEntry.fromBlock(next, cur);
+      await wdb.addBlock(entry, [tx]);
+
+      const rawName = Buffer.from('cloudflare', 'ascii');
+      const nameHash = rules.hashName(rawName);
```

Comment:
You could set `rawName` and `nameHash` at the top of the `describe` block and remove some repeated lines.